tuning/automation/rocblas-parser.py

```diff
@@ -0,0 +1,240 @@
+################################################################################
+#
+# Copyright (C) 2016-2023 Advanced Micro Devices, Inc. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+################################################################################
+
+# reads the output of rocblas-bench with ROCBLAS_LAYER=2, and summarizes the sizes based on GEMM types.
+
+# Usage
+# python3 rocblas-parser.py rocblas-configs.log sizes.log
+
+import os
+import argparse
+
+def gemmfinder(config):
+    output ="unrecognized"
+    # BBS
+    if   (config.f == "gemm_ex" and config.transposeA == "N" and config.transposeB == "N" and config.a_type == "bf16_r" and config.b_type == "bf16_r"  and config.c_type == "bf16_r"  and config.d_type == "bf16_r" and config.compute_type == "f32_r"): 

```

Comment:
Can we simplify the code?
At least, we can add "_NN" or "_NT" or "_TN" or "_TT" separately and unify the code for orientation.

Also, we can add "_SB" separately and unify the code for "_SB"

Personally, I prefer to combine "BB" and "BS" like this.
if config.a_type == "bf16_r" and config.b_type == "bf16_r":
  if config.c_type == "bf16_r" and config.d_type == "bf16_r":
      output = "BB"
  elif config.c_type == "f32_r" and config.d_type == "f32_r":
      output = "BS"